### PR TITLE
sphinx: fixing issue of cython functions not being recognized as pyth…

### PR DESCRIPTION
**Pull request by KaiSzuttor**
…on code, thermostat decorator fix

### CODE REVIEW

**Comment by fweik** [comment]
We could remove the python modules from the docker containers and run pip install requirements.txt to check that they are complete and sufficient to build. (decorator module is missing in docker)

**Comment by mkuron** [comment]
@fweik, then we'd give up testing whether Espresso runs with the distribution (older) package versions. Since our requirements are rather trivial, we shouldn't require everyone to use requirements.txt.

**Comment by mkuron** [comment]
@KaiSzuttor, `decorator.decorator`'s job is pretty trivial. We shouldn't need to add a module and intoduce a dependency just to do that, you can probably do it in five lines yourself.

**Review by hmenke — APPROVED** [review]

File: src/python/espressomd/thermostat.pyx [hunk 1]
```diff
@@ -17,13 +17,14 @@
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 #
 from __future__ import print_function, absolute_import
+from decorator import decorator
```
Comment (mkuron) [inline src/python/espressomd/thermostat.pyx]: `decorator.decorator`'s job is pretty trivial. We shouldn't need to add a module and intoduce a dependency just to do that, you can probably do it in five lines yourself. [`decorator`](https://github.com/micheles/decorator/blob/master/src/decorator.py) uses `compile()`/`exec()` inside, so it's a pretty terrible hack anyway!

**Review by mkuron — CHANGES_REQUESTED** [review]

**Review by fweik — APPROVED** [review]

File: src/python/espressomd/thermostat.pyx [hunk 2]
```diff
@@ -17,37 +17,25 @@
...
-
-    def __call__(self, f):
-        def __f(*args, **kwargs):
-            if (not (thermo_switch in self.thermo_type) and
-                  (thermo_switch != THERMO_OFF)):
-                raise Exception("A different thermostat is already set!")
-            f(*args, **kwargs)
-        return __f
+def AssertThermostatType(*allowedthermostats):
+    def decoratorfunction(function):
```
Comment (hmenke) [inline src/python/espressomd/thermostat.pyx]: Perhaps you could have *not* removed the docstring.

**Review by hmenke — APPROVED** [review]

File: src/python/espressomd/thermostat.pyx (hunk 2 above)
Comment (hmenke) [reply src/python/espressomd/thermostat.pyx]: Sorry, GitHub showed me an outdated file.

**Review by mkuron — APPROVED** [review]
Does this also work in Python 2.7? I think `functools.wraps` didn't treat docstrings until Python ~3.5.

**Comment by KaiSzuttor** [comment]
yes

**Comment by KaiSzuttor** [comment]
done, can be merged

**Pull request merged**